x-pack/plugins/ml/public/application/components/saved_objects_warning/saved_objects_warning.tsx

```diff
@@ -61,22 +82,34 @@ export const SavedObjectsWarning: FC<Props> = ({ jobType }) => {
         iconType="alert"
         data-test-subj="mlJobSyncRequiredWarning"
       >
-        <div>
+        <>
           <FormattedMessage
             id="xpack.ml.jobsList.missingSavedObjectWarning.description"
-            defaultMessage="Some jobs are missing their saved object and require synchronization in {link}."
-            values={{
-              link: (
-                <EuiLink href={`${basePath.get()}/app/management/insightsAndAlerting/jobsListLink`}>
-                  <FormattedMessage
-                    id="xpack.ml.jobsList.missingSavedObjectWarning.linkToManagement.link"
-                    defaultMessage="Stack Management"
-                  />
-                </EuiLink>
-              ),
-            }}
+            defaultMessage="Some jobs are missing or have incomplete saved objects. "
           />
-        </div>
+          {canCreateJob ? (
+            <FormattedMessage
+              id="xpack.ml.jobsList.missingSavedObjectWarning.link"
+              defaultMessage=" {link}"
+              values={{
+                link: (
+                  <EuiLink onClick={() => setShowSyncFlyout(true)}>
+                    <FormattedMessage
+                      id="xpack.ml.jobsList.missingSavedObjectWarning.linkToManagement.link"
+                      defaultMessage="Synchronize your jobs."
+                    />
+                  </EuiLink>
+                ),
+              }}
+            />
+          ) : (
+            <FormattedMessage
+              id="xpack.ml.jobsList.missingSavedObjectWarning.noPermission"
+              defaultMessage="Please synchronize your jobs in Stack Management."
```

Comment:
If I'm not mistaken, with Kibana writing guidelines we should avoid using "Please" 🤔 cc @szabosteve

Reply:
Yes, according to the [guideline](https://elastic.github.io/eui/#/guidelines/writing#messages) in most cases, "please" is unnecessary. Exceptions are situations where the user must wait or do something inconvenient. For example, switch context or going to another page. If it's a warning, then it is not necessary to include "please".

Reply:
updated in 00ce0f7